lib/ecs_deploy/auto_scaler.rb

```diff
@@ -406,9 +406,8 @@ def detach_and_terminate_orphan_instances(service_config)
       def running_essential_task?(instance, service_config)
         return false if instance.running_tasks_count == 0
 
-        task_arns = service_config.client.list_tasks(cluster: service_config.cluster, container_instance: instance.container_instance_arn).task_arns
-        task_groups = service_config.client.describe_tasks(cluster: service_config.cluster, tasks: task_arns).tasks.map(&:group)
-        task_groups.include?("service:#{service_config.name}")
+        container_arns = service_config.client.list_container_instances(cluster: service_config.cluster, filter: "task:group == service:#{service_config.name}")[0]
+        container_arns.include?(instance.container_instance_arn)
```

Comment:
I confirmed that this diff return the same results.

The test environment like this:

- 1 cluster
  - 1 autoscaling group for the cluster
- 2 services in the cluster
- 1 service is specified with `services:name` key in YAML config file for autoscaler
  - then I stop and run this specified service's task and check result

detail for test environment: https://gist.github.com/woshidan/5463dfeba59cb506c3ca2cc4d2106e4a